lib/IRGen/Field.h

```diff
@@ -0,0 +1,107 @@
+//===--- Field.h - Abstract stored field ------------------------*- C++ -*-===//
+//
+// This source file is part of the Swift.org open source project
+//
+// Copyright (c) 2014 - 2017 Apple Inc. and the Swift project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See https://swift.org/LICENSE.txt for license information
+// See https://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+//
+//===----------------------------------------------------------------------===//
+//
+//  This file provides an abstraction for some sort of stored field
+//  in a type.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef SWIFT_IRGEN_FIELD_H
+#define SWIFT_IRGEN_FIELD_H
+
+#include "swift/AST/Decl.h"
+
+namespace swift {
+class SILType;
+
+namespace irgen {
+class IRGenModule;
+
+/// Return the number of fields that will be visited by forEachField,
+/// which determines a number of things in the ABI, including the length
+/// of the field vector in the type metadata.
+///
+/// Generally this is the length of the stored properties, but 
+/// root default actors have an implicit field for their default-actor
+/// storage, and there may also be missing members.
+///
+/// Even if you're sure that you're in a case where only the stored
+/// properties are needed, calling this (and forEachField) instead of
+/// directly accessing the stored properties is good practice.
+unsigned getNumFields(const NominalTypeDecl *type);
+
+struct Field {
+public:
+  enum Kind: uintptr_t {
+    Var,
+    MissingMember,
+    DefaultActorStorage,
+    FirstArtificial = DefaultActorStorage
```

Comment:
Could you add a short description for what 'artificial' means in this context/why the actor storage can't be represented as an ordinary var?